include/triton/Conversion/TritonGPUToLLVM/ElementwiseOpToLLVM.h

```diff
@@ -0,0 +1,348 @@
+#ifndef TRITON_CONVERSION_TRITONGPU_TO_ELEMENTWISE_OP_H
+#define TRITON_CONVERSION_TRITONGPU_TO_ELEMENTWISE_OP_H
+
+#include "mlir/Conversion/LLVMCommon/Pattern.h"
+#include "mlir/Conversion/LLVMCommon/TypeConverter.h"
+#include "triton/Analysis/AxisInfo.h"
+#include "triton/Conversion/TritonGPUToLLVM/Utility.h"
+
+using namespace mlir;
+using namespace mlir::triton;
+
+namespace mlir::triton {
+
+namespace gpu {
+
+inline Type getElementType(Value value) {
+  auto type = value.getType();
+  if (auto tensorType = type.dyn_cast<RankedTensorType>())
+    return tensorType.getElementType();
+  return type;
+}
+
+// MMA encoding has a different order depending on the element's bit width;
+// reorder if we're in this case.
+static SmallVector<Value> reorderValues(const SmallVector<Value> &values,
```

Comment:
does all this code needs to be in the header? reorderValues is only used by nvidia backend right? Can the rest of the code be implemented in a cpp file

Reply:
Ah, yes! I moved the definition to a cpp file, and included that in the header. I renamed the files to `Elementwise...Base.h/cpp`. As for `reorderValues`, it doesn't seems to be nvidia specific. The function description confused me too, but I do see it being used in the template base class (which is used for both nvidia/amd conversions) and in `ElementwiseInlineAsmOpConversion`, which seems to lower to `LLVM::InlineAsmOp`, which I assumed is backend agnostic. Let me know if I'm wrong about this.

Now I also see that `Utility.h` (which now lives in `include/`) also has function definitions, and I'm not sure if all these functions are needed in both backends, but I'll leave this for another PR.

Reply:
`reorderValues` is specific to nvidia mma layouts, therefore it should be needed on AMD's side. It might require removing more pieces in the base class. 
Cleaning this up in a follow up PR sounds good.